src/js/bookmarkList.js

```diff
@@ -35,12 +27,25 @@ window.angular.module('pocket', [
 
   $scope.bookmarks = [];
   $scope.allResultsFetched = false;
+  $scope.pagePocketed = false;
 
   $scope.$watch('bookmarks', function(newVal, oldVal) {
     if (newVal !== oldVal && newVal === []) {
       $scope.allResultsFetched = false;
       $scope.loadNextPage();
     }
+    // Change add button to article view if page has already been pocketed
+    // or back to add button if it has been removed
+    common.getActiveTab().then(function(tab) {
+      var item = _.findWhere($scope.bookmarks, {url: tab.url});
+      if (item && !$scope.pagePocketed) {
```

Comment:
What used to be just an "add this page" button now does double duty. When the page is already in pocket, it turns into an "Article View" button (see https://getpocket.com/developer/docs/v3/article-view). So we need to keep track of whether the page was already in Pocket so we can tell whether we need to change the button.